requires.py

```diff
@@ -28,19 +28,22 @@ def changed(self):
         # assume we have all the data if we have the port
         if conv.get_remote('port'):
             conv.set_state('{relation_name}.available')
-        
+
     @hook('{requires:elasticsearch}-relation-{broken, departed}')
     def broken(self):
         self.remove_state('{relation_name}.connected')
         self.remove_state('{relation_name}.connected')
         self.set_state('{relation_name}.broken')
 
+    def list_unit_data(self):
+        ''' Iterate through all ElasticSearch conversations and return the data
+        for each cached conversation. This allows us to build a cluster string
+        directly from the relation data. eg:
 
-
-    def cluster_name(self):
-        self.get_remote('cluster-name')
-
-    def list_units(self):
+        for unit in elasticsearch.list_data():
+            print(unit['cluster_name'])
+        '''
         for conv in self.conversations():
-            unit = conv.scope
-            yield unit
+            yield {'cluster_name': conv.get_remote('cluster_name'),
```

Comment:
this comes from the remote data.

So as I understand it, and feel free to correct me @johnsca  or @marcoceppi :+1: 

Interface layers cache data that is explicitly sent across the wire with relation-set, the auto_accessors are convenience method(s) for globally scoped relationship interfaces. This is a scope:unit so we need to iterate the conversation objects to retrieve the data, similar to how we used a scope={{conv scope}} while setting relation data on a non-globally scoped relation.

So, in short, the data is coming from elasticsearch, which is relation-setting the data. we dont get handy accessors because the scope:unit of the relationship doesn't retain that self. bit, unless its implicitly in the interface class, during the hook sequence that the conversation is assigned to self.

or something along those lines.